File: data/src/main/java/org/cryptomator/data/cloud/crypto/RootCryptoFolder.java
package org.cryptomator.data.cloud.crypto;

import org.cryptomator.domain.Cloud;

class RootCryptoFolder extends CryptoFolder {

	private final CryptoCloud cloud;

	public RootCryptoFolder(CryptoCloud cloud) {
		super(null, "", "", null);
		this.cloud = cloud;
	}

	public static boolean isRoot(CryptoFolder folder) {
		return folder instanceof RootCryptoFolder;
	}

	@Override
	public Cloud getCloud() {
		return cloud;
	}

	@Override
	public CryptoFolder withCloud(Cloud cloud) {
		return new RootCryptoFolder((CryptoCloud) cloud);
	}
}

File: data/src/main/java/org/cryptomator/data/cloud/crypto/RootCryptoFolder.kt
package org.cryptomator.data.cloud.crypto

import org.cryptomator.domain.Cloud

class RootCryptoFolder(override val cloud: CryptoCloud) : CryptoFolder(null, "", "", null) {

	override fun withCloud(cloud: Cloud?): CryptoFolder {
		return RootCryptoFolder(cloud as CryptoCloud)
	}

	companion object {

		@JvmStatic
		fun isRoot(folder: CryptoFolder): Boolean {
			return folder is RootCryptoFolder
		}
	}
}
